src/merkle/simple_smt/mod.rs

```diff
@@ -15,14 +15,20 @@ pub struct SimpleSmt {
     depth: u8,
     root: Word,
     leaves: BTreeMap<u64, Word>,
-    pub(crate) branches: BTreeMap<NodeIndex, BranchNode>,
+    branches: BTreeMap<NodeIndex, BranchNode>,
     empty_hashes: Vec<RpoDigest>,
 }
 
 #[derive(Debug, Default, Clone, PartialEq, Eq)]
-pub(crate) struct BranchNode {
-    pub(crate) left: RpoDigest,
-    pub(crate) right: RpoDigest,
+struct BranchNode {
+    left: RpoDigest,
```

Comment:
question: why is the Smt and [`MerkleStore`](https://github.com/0xPolygonMiden/crypto/blob/next/src/merkle/store/mod.rs#L73) using `RpoDigest` instead of `Word`? meanwhile [`MerklePath`](https://github.com/0xPolygonMiden/crypto/blob/next/src/merkle/path.rs#L10) and as a consequence `MerklePathSet`, [`MerkleTree`](https://github.com/0xPolygonMiden/crypto/blob/next/src/merkle/merkle_tree.rs#L15), [`Mmr`](https://github.com/0xPolygonMiden/crypto/blob/next/src/merkle/mmr/full.rs#L39) use `Word`, which adds to the number of conversions we have to do (it is easy to do, just unexpected).

Reply:
`MerkleStore` needs to use `RpoDigest` because it can be used as a key in a `BTreeMap` while a `Word` can't. I think the solution here is to change `Word` to `RpoDigest` in other structure as well - but probably should be done in another PR.

Reply:
ah, I always forget we don't have equality for Word, thanks!